hippy/module/pypy_bridge/py_strategies.py

```diff
@@ -97,6 +97,33 @@ def pop(self, w_list, index):
 
         return w_php_val.to_py(interp)
 
+    def getslice(self, w_list, start, stop, step, length):
+        """Adapted from the PyPy implementation"""
+
+        interp = self.space.get_php_interp()
+        w_php_arry_ref = self.unerase(w_list.lstorage)
+        w_php_arry = w_php_arry_ref.deref_temp()
+
+        if step == 1 and 0 <= start <= stop:
+            assert start >= 0
+            assert stop >= 0
+            sublist = w_php_arry._get_raw_storage()[start:stop]
+            sublist_py = [x.to_py(interp) for x in sublist]
+
+        else:
+            sublist_py = [self._none_value] * length
+            for i in range(length):
+                try:
+                    sublist_py[i] = \
+                        w_php_arry._get_raw_storage()[start].to_py(interp)
+                    start += step
+                except IndexError:
+                    raise
+
+        # desired strategy is unknown, so create via objspace
+        return self.space.newlist(sublist_py)
+
+
```

Comment:
Indeed. I did consider this and came to the following conclusion: since slicing is a copying operation, and since the copy is initiated in Python code, the new list is born as a Python list.

Of course, it could go the other way around. What do you think?